notebook/static/tree/js/notebooklist.js

```diff
@@ -697,6 +707,71 @@ define([
         });
     };
 
+    NotebookList.prototype.move_selected = function() {
+        var that = this;
+
+        // TODO(nhdaly): Support moving multiple items at once.
+        if (that.selected.length !== 1){
+            return;
+        }
+
+        var item_path = that.selected[0].path;
+        var item_name = that.selected[0].name;
+        var item_type = that.selected[0].type;
+
+        // Open a dialog to enter the new path, with current path as default.
+        var input = $('<input/>').attr('type','text').attr('size','25').addClass('form-control')
+            .val(utils.url_path_join('/', that.notebook_path));
+        var dialog_body = $('<div/>').append(
+            $("<p/>").addClass("rename-message")
+                .text('Enter new destination directory path for '+ item_type + ':')
+        ).append(
+            $("<br/>")
+        ).append(input);
+        var d = dialog.modal({
+            title : "Move "+ item_type,
+            body : dialog_body,
+            buttons : {
+                Cancel : {},
+                OK : {
+                    class: "btn-primary",
+                    click: function() {
+                      // Construct the new path using the user input and its name.
+                        var new_path = utils.url_path_join(input.val(), item_name)
+                        that.contents.rename(item_path, new_path).then(function() {
+                            that.load_list();
+                        }).catch(function(e) { 
+                            dialog.modal({
+                                title: "Move Failed",
+                                body: $('<div/>')
+                                    .text("An error occurred while moving \"" + item_name + "\" from \"" + item_path + "\" to \"" + new_path + "\".")
+                                    .append($('<div/>')
+                                        .addClass('alert alert-danger')
+                                        .text(e.message || e)),
+                                buttons: {
+                                    OK: {'class': 'btn-primary'}
+                                }
+                            });
+                            console.warn('Error durring moving :', e);
+                        });
+                    }
+                }
+            },
+            open : function () {
+                // Upon ENTER, click the OK button.
+                input.keydown(function (event) {
+                    if (event.which === keyboard.keycodes.enter) {
+                        d.find('.btn-primary').first().click();
+                        return false;
+                    }
+                });
+                input.focus();
+                // Highlight the current path in the input box.
+                input.select();
```

Comment:
Part of what is cool about the Github approach is that as you type paths
with / it detects that and moves those into the path part. Really slick.

On Wed, Jan 20, 2016 at 8:58 AM, Safia Abdalla <notifications@github.com>
wrote:

> In notebook/static/tree/js/notebooklist.js
> <https://github.com/jupyter/notebook/pull/941#discussion_r50282784>:
>
> > +                            console.warn('Error durring moving :', e);
> > +                        });
> > +                    }
> > +                }
> > +            },
> > +            open : function () {
> > +                // Upon ENTER, click the OK button.
> > +                input.keydown(function (event) {
> > +                    if (event.which === keyboard.keycodes.enter) {
> > +                        d.find('.btn-primary').first().click();
> > +                        return false;
> > +                    }
> > +                });
> > +                input.focus();
> > +                // Highlight the current path in the input box.
> > +                input.select();
>
> Hm. Now that I think about it @ellisonbg <https://github.com/ellisonbg>'s
> recommendation is a good one and you won't have to steer too far from the
> code that you have currently written to make it happen.
>
> Here <https://help.github.com/articles/moving-a-file-to-a-new-location/>
> is the help page for GitHub's move feature. So the way I imagine the user
> experience for this working is as follows.
>
>    1. User selects a checkbox on a file in the file tree.
>    2. User clicks the move button in the toolbar above.
>    3. Filename in the file tree gets converted into a file path relative
>    to the directory the user is in.
>    4. User writes within the textbox the new path.
>    5. User hits enter and the file is moved to a new location if there is
>    no error.
>
> Thoughts on this, @NHDaly <https://github.com/NHDaly>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/941/files#r50282784>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com